packages/@headlessui-react/src/components/disclosure/disclosure.tsx

```diff
@@ -73,17 +71,17 @@ enum ActionTypes {
   SetButtonId,
   SetPanelId,
 
-  LinkPanel,
-  UnlinkPanel,
+  SetButtonElement,
+  SetPanelElement,
 }
 
 type Actions =
   | { type: ActionTypes.ToggleDisclosure }
   | { type: ActionTypes.CloseDisclosure }
   | { type: ActionTypes.SetButtonId; buttonId: string | null }
   | { type: ActionTypes.SetPanelId; panelId: string | null }
-  | { type: ActionTypes.LinkPanel }
-  | { type: ActionTypes.UnlinkPanel }
+  | { type: ActionTypes.SetButtonElement; element: HTMLButtonElement | null }
+  | { type: ActionTypes.SetPanelElement; element: HTMLElement | null }
 
```

Comment:
We already tracked some state for linking the panel. Thinking about this, this was actually a bit of a smell because it forced a re-render so that we knew about the `ref.current` being filled in...

I also renamed it to be consistent with other components.